go/private/sdk.bzl

```diff
@@ -177,11 +177,30 @@ def _remote_sdk(ctx, urls, strip_prefix, sha256):
     if len(urls) == 0:
         fail("no urls specified")
     ctx.report_progress("Downloading and extracting Go toolchain")
-    ctx.download_and_extract(
-        url = urls,
-        stripPrefix = strip_prefix,
-        sha256 = sha256,
-    )
+    if urls[0].endswith(".tar.gz"):
+        # BUG(#2771): Use a system tool to extract the archive instead of
+        # Bazel's implementation. With some configurations (macOS + Docker +
+        # some particular file system binding), Bazel's implementation rejects
+        # files with invalid unicode names. Go has at least one test case with a
```

Comment:
Is there a bazel bug to track this?

Reply:
Not at the moment. It seemed like a pretty intricate configuration, so I didn't try to reproduce it. I'd rather not report it upstream though without detailed instructions, since it's very unlikely to get fixed.

(Sorry I realize this sounds lazy; it's a consequence of not being able to spend adequate time on this).

Reply:
I _think_ the Bazel bug is https://github.com/bazelbuild/bazel/issues/12986